prive/datasets/dataset.py

```diff
@@ -63,76 +65,210 @@ def replace(self, record_in, record_out):
 
-    def drop_record(self, record_ids):
         """
-        Drop a record(s) and return modified dataset.
+
+        # TODO: what if the index is supposed to be a column? an identifier?
+        return TabularDataset(self.dataset.iloc[record_ids], self.description)
+
+    def drop_records(self, record_ids):
         """
-        pass
+        Get a record from the TabularDataset object
+
+        Parameters
+        ----------
+        record_ids (List[int]): List of indexes of records to drop
 
+        Returns
+        -------
 
-    def add_record(self, record):
+        A TabularDataset object without the record(s).
+
+        """
+        # TODO: what if the index is supposed to be a column? an identifier?
+        return TabularDataset(self.dataset.drop(record_ids), self.description)
+
+    def add_records(self, records):
         """
         Add record(s) to dataset and return modified dataset.
+
+        Parameters
+        ----------
+        records (TabularDataset): A TabularDataset object with the record(s) to add.
+
+        Returns
+        -------
+
+        A new TabularDataset object with the record(s).
         """
-        pass
 
-    def replace(self, record_in, record_out):
+        # this does the same as the __add__
+        return self.__add__(records)
+
+    def replace(self, records_in, records_out):
         """
-        Replace a row with a given row.
+
+        Replace a record with another one in the dataset.
+
+        Parameters
+        ----------
+        records_in (TabularDataset): A TabularDataset object with the record(s) to add.
+        records_out (List(int)): List of indexes of records to drop
+
+        Returns
+        -------
+
+        A modified TabularDataset object with the replaced record(s).
+
+
         """
-        pass
 
-    def create_subsets(self, n, sample_size, record_drop=None):
-        """ Create a number of training datasets (sub-samples from main dataset)
-         of a given sample size and with the option to remove a recotd"""
+        reduced_dataset = self.drop_records(records_out)
+
+        return reduced_dataset.add_records(records_in)
+
         pass
 
+    def create_subsets(self, n, sample_size, target_records):
+        """
+        Create a number of training datasets (sub-samples from main dataset)
+         of a given sample size  with and without target records.
+
+        Parameters
+        ----------
+        n (int): Number of datasets to create.
+        sample_size (int) : Size of the subset datasets to be created
+        target_records (List(int)): List of indexes of the target records.
+
+        Returns
+        -------
+
+        Two List(TabularDataset) containing subsets of the data with and without the target record(s).
+
+        """
+        # remove target records
+        dataset_notarget = self.drop_records(target_records)
+
+        # create splits
+        kf = ShuffleSplit(n_splits=n, train_size=sample_size)
```

Comment:
I re-implemented a util for generating indices, because ShuffleSplit didn't seem to quite do what I was hoping. I think we should leave target inclusion/exclusion logic to the `ThreatModel` class, so that this method just generates subsets, and could optionally take a `drop_targets` argument (though I think that other code using this class would be cleaner to explicitly drop the targets).

Reply:
i modified the function to only generate subsets, i have used your index_split function.